src/lib.rs

```diff
@@ -155,6 +157,55 @@ pub fn clean_text(src: &str) -> String {
     ret_val
 }
 
+/// Determine if a given string contains HTML
+///
+/// This function is parses the full string into HTML and checks if the input contained any
+/// HTML syntax.
+///
+/// # Note
+/// This function cannot will return positively for strings that contain invalid HTML syntax like
+/// `<g>` and even `Vec::<u8>::new()`.
+pub fn is_html(input: &str) -> bool {
+    let santok = SanitizationTokenizer::new();
+    let mut chunk = ByteTendril::new();
+    chunk.push_slice(input.as_bytes());
+    let mut input = BufferQueue::new();
+    input.push_back(chunk.try_reinterpret().unwrap());
+
+    let mut tok = Tokenizer::new(santok, Default::default());
+    let _ = tok.feed(&mut input);
+    tok.end();
+    tok.sink.was_sanitized
+}
+
+#[derive(Copy, Clone)]
+struct SanitizationTokenizer {
+    was_sanitized: bool,
+}
+
+impl SanitizationTokenizer {
+    pub fn new() -> SanitizationTokenizer {
+        SanitizationTokenizer {
+            was_sanitized: false,
+        }
+    }
+}
+
+impl TokenSink for SanitizationTokenizer {
+    type Handle = ();
+    fn process_token(&mut self, token: Token, _line_number: u64) -> TokenSinkResult<()> {
+        println!("Heh, token! {:?}", token);
```

Comment:
This probably shouldn't be here. It's for your debugging purposes, right?

Reply:
oops. well spotted.